app/lib/crypto/address/p2pkh.ts

```diff
@@ -0,0 +1,15 @@
+import {PublicKey} from "../key/publicKey"
+import * as Bech32 from "bech32"
+import {sha256} from "../hash"
+
+/**
+ * Encode address
+ * @param {PublicKey} pubKey public key
+ * @param {"wit" | "twit"} hrp human readable part
+ * @returns {string} address
+ */
+export const encode = (pubKey: PublicKey, hrp: "wit" | "twit"): string => {
```

Comment:
Using strings for the hrp argument would only make sense if this function accepted any arbitrary value, for example in the case that we wanted to support other blockchains with different prefixes.

The problem is that we will need to hardcode the same string in many different places across our project, and use something like `encode(publicKey, app.chain === "main" ? "wit" : "twit")` pretty much everywhere. This is suboptimal because it introduces a lot of code repetition and diminish flexibility if we want to change the prefixes at some point.

Instead, I'd suggest `encode(publicKey, app.chain)` where `app.chain` has the following enum type:
```ts
enum ChainType {
  "test",
  "main",
}
```
Then the `encode/2` function should internally map the chain type to an hrp:
```ts
enum Prefix {
    "twit",
    "wit"
}

function encode(pubKey: PublicKey, chain: ChainType): string {
    if (!(chain in Prefix)) { 
        throw new Error("Unsupported chain type")
    }

    const hrp = Prefix[chain]

    /* ... */
}

encode(key, ChainType.main)
```